docs/pbc/all/payment-service-provider/202410.0/base-shop/payment-method-strategies.md

```diff
@@ -67,6 +67,55 @@ The `scripts` array contains the URLs to the scripts that are needed to display
 
 The `payment_service_provider_data` contains the data that can be used by projects to configure the Payment Method. Those value may differ for each different Payment Method.
 
+### Query Params for Scripts
+
+It is important to understand who is responsible for what. The App itself knows only which query params (keys) are needed. The Shop knows which values are needed and where to get them from.  
+
+For example the script must be like `https://example.com/js?currency=EUR`. The App knows that the `currency` is needed but can't know which one is used in the Shop. The Shop knows that the used `currency` is `EUR`.
+
+The App defines a Scrtipt like this
+
+```json
+{
+   "scripts": [
+      {
+         "url": "https://example.com/js",
+         "query_params": {
+            "currency": "currencyName"
+         }
+      }
+   ]
+}
+```
+
+The Shop loads this information and parses the `currencyName` to `EUR` and the final URL is `https://example.com/js?currency=EUR`.
+
+Why not simpler and only using a key? The answer is easy aURL could look like `https://example.com/js?c=EUR` where the key can't be used anymore to determine what the value should be.
+
+In this example the App would define the script like this
+
+```json
+{
+   "scripts": [
+      {
+         "url": "https://example.com/js",
+         "query_params": {
+            "c": "currencyName"
+         }
+      }
+   ]
+}
+```
+
+So that the Shop can understand that `c` has to be used as param key in the URL and the value is the name of the currency `currencyName`.
+
+Current on SCOS side supported query params:
+
+- `currencyName`
+- `localeName`
```

Comment:
in merchant-app (for merchant portal) we already have 

 - `_merchantReference_`
 - `_locale_`


https://github.com/spryker/merchant-app-merchant-portal-gui/blob/4b5daf6c5f9b1471ec0667f2175a197655a93bd5/src/Spryker/Zed/MerchantAppMerchantPortalGui/Communication/Controller/PaymentSettingsController.php#L199

I would use the same format.

Reply:
That is kinda different, in your example, it's a replace of __placeholder__ on a URL where as in the current one we have a name that acts as a key which would look weird.